// @azure/cognitiveservices-luis-runtime/esm/models/mappers.js
/*
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License. See License.txt in the project root for
 * license information.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
export var Intent = {
    serializedName: "Intent",
    type: {
        name: "Composite",
        className: "Intent",
        modelProperties: {
            score: {
                serializedName: "score",
                type: {
                    name: "Number"
                }
            },
            childApp: {
                serializedName: "childApp",
                type: {
                    name: "Composite",
                    className: "Prediction"
                }
            }
        }
    }
};
export var Sentiment = {
    serializedName: "Sentiment",
    type: {
        name: "Composite",
        className: "Sentiment",
        modelProperties: {
            label: {
                serializedName: "label",
                type: {
                    name: "String"
                }
            },
            score: {
                required: true,
                serializedName: "score",
                type: {
                    name: "Number"
                }
            }
        }
    }
};
export var Prediction = {
    serializedName: "Prediction",
    type: {
        name: "Composite",
        className: "Prediction",
        modelProperties: {
            alteredQuery: {
                serializedName: "alteredQuery",
                type: {
                    name: "String"
                }
            },
            topIntent: {
                required: true,
                serializedName: "topIntent",
                type: {
                    name: "String"
                }
            },
            intents: {
                required: true,
                serializedName: "intents",
                type: {
                    name: "Dictionary",
                    value: {
                        type: {
                            name: "Composite",
                            className: "Intent"
                        }
                    }
                }
            },
            entities: {
                required: true,
                serializedName: "entities",
                type: {
                    name: "Dictionary",
                    value: {
                        type: {
                            name: "Object"
                        }
                    }
                }
            },
            sentiment: {
                serializedName: "sentiment",
                type: {
                    name: "Composite",
                    className: "Sentiment"
                }
            }
        }
    }
};
export var PredictionResponse = {
    serializedName: "PredictionResponse",
    type: {
        name: "Composite",
        className: "PredictionResponse",
        modelProperties: {
            query: {
                required: true,
                serializedName: "query",
                type: {
                    name: "String"
                }
            },
            prediction: {
                required: true,
                serializedName: "prediction",
                type: {
                    name: "Composite",
                    className: "Prediction"
                }
            }
        }
    }
};
export var ErrorBody = {
    serializedName: "ErrorBody",
    type: {
        name: "Composite",
        className: "ErrorBody",
        modelProperties: {
            code: {
                required: true,
                serializedName: "code",
                type: {
                    name: "String"
                }
            },
            message: {
                required: true,
                serializedName: "message",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var ErrorModel = {
    serializedName: "Error",
    type: {
        name: "Composite",
        className: "ErrorModel",
        modelProperties: {
            error: {
                required: true,
                serializedName: "error",
                type: {
                    name: "Composite",
                    className: "ErrorBody"
                }
            }
        }
    }
};
export var PredictionRequestOptions = {
    serializedName: "PredictionRequestOptions",
    type: {
        name: "Composite",
        className: "PredictionRequestOptions",
        modelProperties: {
            datetimeReference: {
                serializedName: "datetimeReference",
                type: {
                    name: "DateTime"
                }
            },
            preferExternalEntities: {
                serializedName: "preferExternalEntities",
                type: {
                    name: "Boolean"
                }
            }
        }
    }
};
export var ExternalEntity = {
    serializedName: "ExternalEntity",
    type: {
        name: "Composite",
        className: "ExternalEntity",
        modelProperties: {
            entityName: {
                required: true,
                serializedName: "entityName",
                type: {
                    name: "String"
                }
            },
            startIndex: {
                required: true,
                serializedName: "startIndex",
                type: {
                    name: "Number"
                }
            },
            entityLength: {
                required: true,
                serializedName: "entityLength",
                type: {
                    name: "Number"
                }
            },
            resolution: {
                serializedName: "resolution",
                type: {
                    name: "Object"
                }
            },
            score: {
                serializedName: "score",
                type: {
                    name: "Number"
                }
            }
        }
    }
};
export var RequestList = {
    serializedName: "RequestList",
    type: {
        name: "Composite",
        className: "RequestList",
        modelProperties: {
            name: {
                serializedName: "name",
                type: {
                    name: "String"
                }
            },
            canonicalForm: {
                required: true,
                serializedName: "canonicalForm",
                type: {
                    name: "String"
                }
            },
            synonyms: {
                serializedName: "synonyms",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "String"
                        }
                    }
                }
            }
        }
    }
};
export var DynamicList = {
    serializedName: "DynamicList",
    type: {
        name: "Composite",
        className: "DynamicList",
        modelProperties: {
            listEntityName: {
                required: true,
                serializedName: "listEntityName",
                type: {
                    name: "String"
                }
            },
            requestLists: {
                required: true,
                serializedName: "requestLists",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "RequestList"
                        }
                    }
                }
            }
        }
    }
};
export var PredictionRequest = {
    serializedName: "PredictionRequest",
    type: {
        name: "Composite",
        className: "PredictionRequest",
        modelProperties: {
            query: {
                required: true,
                serializedName: "query",
                type: {
                    name: "String"
                }
            },
            options: {
                serializedName: "options",
                type: {
                    name: "Composite",
                    className: "PredictionRequestOptions"
                }
            },
            externalEntities: {
                serializedName: "externalEntities",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "ExternalEntity"
                        }
                    }
                }
            },
            dynamicLists: {
                serializedName: "dynamicLists",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "DynamicList"
                        }
                    }
                }
            }
        }
    }
};
