// @azure/cognitiveservices-luis-runtime/esm/models/parameters.js
/*
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License. See License.txt in the project root for
 * license information.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */
export var appId = {
    parameterPath: "appId",
    mapper: {
        required: true,
        serializedName: "appId",
        type: {
            name: "Uuid"
        }
    }
};
export var endpoint = {
    parameterPath: "endpoint",
    mapper: {
        required: true,
        serializedName: "Endpoint",
        defaultValue: '',
        type: {
            name: "String"
        }
    },
    skipEncoding: true
};
export var log = {
    parameterPath: [
        "options",
        "log"
    ],
    mapper: {
        serializedName: "log",
        type: {
            name: "Boolean"
        }
    }
};
export var showAllIntents = {
    parameterPath: [
        "options",
        "showAllIntents"
    ],
    mapper: {
        serializedName: "show-all-intents",
        type: {
            name: "Boolean"
        }
    }
};
export var slotName = {
    parameterPath: "slotName",
    mapper: {
        required: true,
        serializedName: "slotName",
        type: {
            name: "String"
        }
    }
};
export var verbose = {
    parameterPath: [
        "options",
        "verbose"
    ],
    mapper: {
        serializedName: "verbose",
        type: {
            name: "Boolean"
        }
    }
};
export var versionId = {
    parameterPath: "versionId",
    mapper: {
        required: true,
        serializedName: "versionId",
        type: {
            name: "String"
        }
    }
};
